.zuul.yaml

```
@@ -0,0 +1,216 @@
+      - name: controller
+        label: cloud-centos-9-stream-tripleo-vexxhost
+      - name: crc
+        label: coreos-crc-extracted-2-19-0-xxl
+
+- nodeset:
+    name: stf-crc_extracted-ocp413
+    nodes:
+      - name: controller
+        label: cloud-centos-9-stream-tripleo-vexxhost
+      - name: crc
+        label: coreos-crc-extracted-2-28-0-xxl
+
+- nodeset:
+    name: stf-crc_extracted-ocp414
+    nodes:
+      - name: controller
+        label: cloud-centos-9-stream-tripleo-vexxhost
+      - name: crc
+        label: coreos-crc-extracted-2-30-0-xxl
+
+# Based on the 2-node job cookbook at https://github.com/openstack-k8s-operators/ci-framework/blob/main/docs/source/cookbooks/zuul-job-nodeset.md
+- job:
+    name: stf-base-2node
+    parent: podified-multinode-edpm-deployment-crc
+    abstract: true
+    required-projects:
+      - name: github.com/openstack-k8s-operators/dataplane-operator
+        override-checkout: main
+      - name: github.com/openstack-k8s-operators/infra-operator
+        override-checkout: main
+      - name: github.com/openstack-k8s-operators/openstack-operator
+        override-checkout: main
+      - name: github.com/openstack-k8s-operators/openstack-must-gather
+        override-checkout: main
+    pre-run:
+      - ci/pre-2node.yml
+    vars:
+      cifmw_deploy_edpm: false
+      podified_validation: true
+      cifmw_run_tests: false
+    extra-vars:
+      crc_ci_bootstrap_networking:
+        networks:
+          default:
+            range: 192.168.122.0/24
+            mtu: 1500
+          internal-api:
+            vlan: 20
+            range: 172.17.0.0/24
+          storage:
+            vlan: 21
+            range: 172.18.0.0/24
+          tenant:
+            vlan: 22
+            range: 172.19.0.0/24
+        instances:
+          controller:
+            networks:
+              default:
+                ip: 192.168.122.11
+          crc:
+            networks:
+              default:
+                ip: 192.168.122.10
+              internal-api:
+                ip: 172.17.0.5
+              storage:
+                ip: 172.18.0.5
+              tenant:
+                ip: 172.19.0.5
+
+- job:
+    name: stf-base
+    # defined in: https://review.rdoproject.org/cgit/config/tree/zuul.d/_jobs-crc.yaml
+    parent: stf-base-2node
+    abstract: true
+    description: |
+      Run the stf-run-ci role, and then test stf
+    roles:  # adds in dependent roles i.e. put it in the role path
+      - zuul: github.com/openstack-k8s-operators/ci-framework
+    # These are the additional repos that zuul will clone
+    required-projects:
+      - name: openstack-k8s-operators/ci-framework
+        override-checkout: main
+      - name: github.com/infrawatch/service-telemetry-operator
+      - name: github.com/infrawatch/smart-gateway-operator
+      - name: github.com/infrawatch/sg-bridge
+      - name: github.com/infrawatch/sg-core
+      - name: github.com/infrawatch/prometheus-webhook-snmp
```

```suggestion
      - name: github.com/infrawatch/smart-gateway-operator
        override-checkout: stable-1.5
      - name: github.com/infrawatch/sg-bridge
        override-checkout: stable-1.5
      - name: github.com/infrawatch/sg-core
        override-checkout: stable-1.5
      - name: github.com/infrawatch/prometheus-webhook-snmp
        override-checkout: stable-1.5
```

@elfiesmelfie can we get your :+1:  on this? Do we need these?

---

Reading zuul docs[1], I think these are not required

"When Zuul runs jobs for a proposed change, it normally checks out the branch associated with that change on every project present in the job"

In fact I think this would be harmful, for example it could fail to test our release-prep-1.5.4 branch because of dependent changes that are not on stable-1.5 yet

[1] https://zuul-ci.org/docs/zuul/latest/config/job.html#attr-job.required-projects.override-checkout

---

IMHO, it's better to be explicit, since I'm not 100% sure which the base branch will be. 

There is a quick way to check which versions Zuul has checked out.
Zuul creates an inventory file that shows the vars set for the job, which is available in the logs as the zuul-info/inventory.yaml:

https://logserver.rdoproject.org/74/574/c7cfcd3e735e03a2ae3d872b337eaa9eb3fbfa0a/github-check/stf-crc-ocp_414-local_build-index_deploy/d39737d/zuul-info/inventory.yaml

I think these SHOULD be pointed at release-prep-1.5.4 right now.
In order to test it properly, I recommend having this pr use stable-1.5 (so that it is testing stable-1.5 AFTER it merges)
For testing the release-prep branch, we may need to open another PR with the override-checkout set to `release-prep-1.5.4`.
This is needed because we don't have any zuul job on the target (stable-1.5) branch.

I am going to check the SHAs for the commits and confirm whether this change is needed.

---

Okay, I'll update the branches.

It might be better to have a follow-up to checkout the stable-1.5 branch, and add release-prep into this one.
I'll update this comment accordingly.

---

TL;DR: it was fine the way it was, not need to add the override-checkout.